core/shared/src/main/scala/com/lynbrookrobotics/potassium/streams/ZippedStream.scala

```diff
@@ -0,0 +1,36 @@
+package com.lynbrookrobotics.potassium.streams
+
+class ZippedStream[A, B](aPeriodicity: ExpectedPeriodicity,
+                         bPeriodicity: ExpectedPeriodicity) extends Stream[(A, B)] {
+  override val expectedPeriodicity: ExpectedPeriodicity = (aPeriodicity, bPeriodicity) match {
+    case (Periodic(a), Periodic(b)) =>
+      if (a == b) {
+        Periodic(a)
+      } else {
+        NonPeriodic
```

Comment:
Wouldn't you want to make it at the `expectedPeriodicity` of the less frequently updating `Steam`?